backend/tracim_backend/lib/webdav/dav_provider.py

```diff
@@ -1,33 +1,140 @@
 from tracim_backend.lib.webdav.lock_storage import LockStorage
+from tracim_backend.lib.webdav.resources import get_content_resource
+from tracim_backend.lib.webdav.resources import get_workspace_resource
+from tracim_backend.models.auth import User
 from tracim_backend.models.data import Content
-from tracim_backend.models.data import ContentNamespaces
 from tracim_backend.models.data import Workspace
 from tracim_backend.models.tracim_session import TracimSession
 
 
+class ResourceLevelType(enum.Enum):
+    ROOT = "root"
+    WORKSPACE = "workspace"
+    CONTENT = "content"
+
+
+class ProcessedWebdavPath(object):
+    """
+    Processor for Webdav Path:
+    - get all workspaces hierarchy from path
+    - get all content hierarchy from path
+    - provide useful properties to handle Webdav Request
+    """
+
+    def __init__(self, path: str, current_user: User, session: TracimSession, app_config: CFG):
+        self.path = path
+        self.workspace_api = WorkspaceApi(
+            current_user=current_user, session=session, config=app_config
+        )
+        self.content_api = ContentApi(current_user=current_user, session=session, config=app_config)
+
+        self.workspaces = deque()
+        self.contents = deque()
+        path_parts = self._path_splitter(self.path)
+        if len(path_parts) >= 1:
+            root_workspace_filemanager_filename = webdav_convert_file_name_to_bdd(path_parts[0])
+            try:
+                root_workspace = self.workspace_api.get_one_by_filemanager_filename(
+                    root_workspace_filemanager_filename
+                )
+            except WorkspaceNotFound:
+                root_workspace = None
+            self.workspaces.append(root_workspace)
+            # INFO - G.M - 2020-10-07 - Get all workspace tree + content_tree
+            if root_workspace:
+                search_sub_workspace = True
+                for path_part in path_parts[1:]:
+                    section = webdav_convert_file_name_to_bdd(path_part)
+                    if search_sub_workspace:
+                        try:
+                            workspace = self.workspace_api.get_one_by_filemanager_filename(
+                                section, parent=self.workspaces[-1]
+                            )
+                            self.workspaces.append(workspace)
+                            continue
+                        except WorkspaceNotFound:
+                            search_sub_workspace = False
+                            pass
+                    try:
+                        content = self.content_api.get_one_by_filename(
+                            filename=section,
+                            workspace=self.workspaces[-1],
+                            parent=self.contents[-1] if self.contents else None,
+                        )
+                    except ContentNotFound:
+                        content = None
+                    self.contents.append(content)
```

Comment:
Wouldn't separating the loop in two parts (one for workspaces, one for contents) be easier to understand?
Something like this:
```python
path_parts = self._path_splitter(self.path)
if not path_parts:
    self.workspaces.append(None)
    return
workspace_found = True
# INFO - G.M - 2020-10-07 - Get all workspace tree + content_tree
while workspace_found:
    try:
        part = path_parts[0]
        filemanager_filename = webdav_convert_file_name_to_bdd(part)
        self.workspaces.append(
            self.workspace_api.get_one_by_filemanager_filename(
                root_workspace_filemanager_filename
            )
        )
        path_parts = path_parts[1:]
    except (IndexError, WorkspaceNotFound):
        workspace_found = False

for part in path_parts:
    try:
        content = self.content_api.get_one_by_filename(
            filename=section,
            workspace=self.workspaces[-1],
            parent=self.contents[-1] if self.contents else None,
        )
    except ContentNotFound:
        content = None
    self.contents.append(content)
```

Reply:
I agree this code is not so nice but i don't really like the idea of changing  dynamically the list used this way.
I suggest to put a comment about finding a way to refactor this better. I try some things but this part is complex so i don't find a proper solution.

Reply:
Updated my suggestion to avoid changing the parts list.

Reply:
Updated with suggestion (with some fix to make it work), i do keep a comment about fact that this mecanism does work but it's not so clear for distinction of some case like invalid path, proper move path and root.